### PR TITLE
Move `VideoCallClient` into its own crate

### PR DESCRIPTION
**Pull request by ronen**
This is a continuation of #137.    It's the same code, but takes it to the next step:   Crate-ified the `VideoCallClient` stuff:  Moved all the code from `yew-ui/src/model` into a new top-level workspace & crate `videocall-client`.   And added documentation!    And of course also conformed `yew-ui` to use the new crate.

For ease of seeing what's going on here, I've based this PR off the `client` branch.    Dunno whether you'd want to merge this onto that branch for merging as part of #137, or I can change this to be based off main and merging this would supercede #137, or wait until #137 is merged on its own then I can rebase all this off of `main` and update this PR.  I have no strong opinions.  But it's done and working so figured I'd put it out for eyeballs now :)

### CODE REVIEW

**Comment by darioalessandro** [comment]
Ok, I'll take a look, thanks for this huge contribution @ronen !!

**Comment by darioalessandro** [comment]
@ronen @griffobeid I think that you guys should talk about https://github.com/security-union/videocall-rs/pull/144

**Review by griffobeid — APPROVED** [review]
This looks great, thanks for adding all of the awesome documentation!

File: videocall-client/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# videocall-client
+
+Version: 0.1.0
+
+---
+
+This crate provides a client-side (browser) interface to the videocall protocol.  The purpose is to
```
[the remark this answers is not in the document]
Comment (ronen) [reply videocall-client/README.md:21]: Yeah, good point.   Added.

**Pull request merged**